### DISCUSSION TITLE
New Pre-release 2.0.2 for SB2 home load and schedule support

### DISCUSSION THREAD

**Discussion by thomluther** [thomluther/ha-anker-solix#105] [Announcements]
### Breaking changes:
- None

### Enhancements:
- Full Support for Solarbank 2 schedule modifications to complete enhancements for Solarbank 2 #60
  - The system output preset entity is now also available for Solarbank 2 devices when the integration configuration uses the Anker owner account
    - It can be used to adjust the SB2 output directly when the `Manual` usage mode is active
  - In the backend, any adjustment of the preset or the usage mode will always result in 2 api requests to update and verify the required schedule structure on the SB2 via the cloud api
    - Subsequent value increases either directly or via services (now actions) will therefore have a cool down of 30 seconds to avoid uncontrolled stress to the api by UI usage or HA automation (identically to the schedule support implementation for SB1)
  - All schedule services (now actions) have been enhanced with an optional `week_day` field to support SB2 time interval adjustments
    - The new field is ignored for SB1 schedule services, like the SB1 specific fields are ignored for SB2 schedule services (actions)
  - The resulting changes and usage implications are described in the [INFO](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md)
- Updated [README ](https://github.com/thomluther/hacs-anker-solix/blob/main/README.md) and [INFO](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md) with changes for Solarbank 2 schedule support and added further references for related blogs and videos
  - Thanks to @TheRealSimon42 for his first contribution and providing additional resources for usage of the integration
  - Thanks also to @thealkly for his previous contributions to the repo and his videos for usage of the integration

### Fixes:
- Adjusted home load helper methods for SB1 and SB2 to prevent existing gaps in the schedule are automatically filled up when inserting or adjusting a slot that has start and end times provided.
- Bump ruff from 0.5.5 to 0.5.6 #103 

### Note:
Since there have been more and more questions on how to integrate the solarbank into the HA energy dashboard (#100 #104), a reference to the corresponding discussion post #16 has been added to the resources in [README ](https://github.com/thomluther/hacs-anker-solix/blob/main/README.md).
I dot **NOT** recommended to use the Anker Solix integration daily energy statistic sensors directly in your energy dashboard, even if that is now possible since they are classified as total_increasing sensors. The reasons for this are described in the discussion [how to integrate the solarbank into you energy dashboard](https://github.com/thomluther/hacs-anker-solix/discussions/16#discussioncomment-10285726). There you can also find the recommended approach for creating the required entities for easiest and most flexible energy dashboard integration.

<hr /><em>This discussion was created from the release <a href='https://github.com/thomluther/hacs-anker-solix/releases/tag/'>2.0.2</a>.</em>

**Comment by thomluther** [comment]
For installation of the Pre-Release 2.0.2, please see instructions in previous pre-release #93 

Please provide feedback if all is working well with the SB2 schedule support since I don't have capabilities to verify the schedule changes with real appliances. In particular:
1. Can the usage mode be changed between manual and Smart Meter mode, and only when the smart meter is installed?
2. Can the usage mode be switched without affecting the schedule?
3. Does an output preset change the applied home load immediately when usage mode is manual? How long does it take for the SB2 to apply the home load change for small changes (100-200 W) and large changes (800 W range)?
4. Can you apply a new time slot plan for specified weekdays with the Set Solarbank schedule service/action? Verify various parameter options and combinations
5. Can you insert or adjust existing time slot with the Update solarbank schedule service/action as described in the INFO usage? Verify various parameter options and combinations
6. Can you delete a time slot plan for individual weekdays with the Clear solarbank schedule service? 
7. Does the SB2 support schedule plans where not all weekdays are specified? Unspecified or removed weekdays should not occur in any plan ID and have the default preset of 200 W applied automatically
8. Does the SB2 apply the existing schedule correctly? Meaning will it turn on when plan changes from 0W interval to an output interval in case it was on standby in the evening?

**Reply by Krosty0712** [reply]
1. Yes Mode can be switched, i have the smartmeter installed.
2-8. I don't use schedules or manual mode, I've only had the SB2 for 7 days and I'm going to see what it can do automatically. Sorry


I am missing a sensor for temperature. In the app i can see the temperature of the SB2.

**Reply by thomluther** [reply]
Thx for the basic confirmation. It would really be helpful, if someone can validate 2-8 somehow.
I plan to release the version end of this week.

The temperature is not provided via the cloud Api, at least I have not seen it in any endpoint response. But that does not mean anything, since I cannot explore any SB2 endpoints.

**Comment by thomluther** [comment]
SB2 home loads via services could not be set below 100 W (#107). This limit was removed in the main branch for now (will not cut new pre-release for this). It can be installed with the same update service as described in #93 by specifying `main` for the version.

**Comment by thomluther** [comment]
The final version for this support is released with 2.1.0
